### PR TITLE
Support detecting unknown server and agent config options

### PR DESCRIPTION
**Pull request by ryysud**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

This PR affects the functionality of `spire-server run` and `spire-agent run`.

**Description of change**

The changes are below.

- Separate server-side config struct into another file as well as agent-side
- Upgrade hashicorp/hcl version to merge commit of https://github.com/hashicorp/hcl/pull/275 because there is a bug in the `unusedKeys` feature in current v1.0.0 
- Support detecting unknown server and agent config options

**Which issue this PR fixes**

This PR fixes ~part of~ https://github.com/spiffe/spire/issues/1101.
~To fix the issue completely, the same changes as this PR are required for agent-side.~
I also added the same changes as server-side to agent-side.

### CODE REVIEW

File: test/fixture/config/agent_bad_agent_block.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+agent {
+    unknwon_option1 = "unknwon_option1"
```
Comment (azdagron) [inline test/fixture/config/agent_bad_agent_block.conf]: nit: typo on `unknwon` (copied and pasted throughout the text fixtures)

Comment (ryysud) [reply test/fixture/config/agent_bad_agent_block.conf]: I fixed it with https://github.com/spiffe/spire/pull/1289/commits/fab9ef08e460fca3b1be00c6809ec0e120ae6354 !

**Comment by evan2645** [comment]
Thank you so much for this @ryysud! Huge quality of life improvement

**Comment by ryysud** [comment]
Thank you for your review!

**Pull request merged**